service/storage_service.cc

```diff
@@ -5423,22 +5426,26 @@ void storage_service::init_messaging_service(sharded<service::storage_proxy>& pr
                 // To fix that, we could store in topology a small history of recent CDC generation IDs
                 // (garbage-collected with time) instead of just the last one, and load all of them.
                 // Alternatively, a node would wait for some time before switching to normal state.
-                auto s = ss._db.local().find_schema(db::system_keyspace::NAME, db::system_keyspace::CDC_GENERATIONS_V3);
-                auto key = dht::decorate_key(*s, partition_key::from_singular(*s, curr_cdc_gen_id->id));
-                auto partition_range = dht::partition_range::make_singular(key);
+                //
+                // FIXME: Topology snapshots contain only mutation of current CDC generation data but don't
```

Comment:
"Topology snapshots contain only mutation of current CDC generation data"
But that's not true (after your changes)

Reply:
Fixed.